### PR TITLE
chore: Migrate password validators

### PR DESCRIPTION
**Pull request by ioanabrooks**
#### Description of changes

* Update password validators to conform to Amplify configure changes.
* Re-enable Change Password e2es
* Re-enable Delete User e2es.

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/ui/src/types/authenticator/form.ts [hunk 1]
```diff
@@ -78,18 +78,10 @@ export type DefaultFormFieldOptions = Record<
...
- * @migration `passwordPolicyCharacters` potential values
- */
-export type PasswordPolicyRules =
-  | 'REQUIRES_LOWERCASE'
-  | 'REQUIRES_NUMBERS'
-  | 'REQUIRES_SYMBOLS'
-  | 'REQUIRES_UPPERCASE';
-
-// password setting directly coming from Amplify.Auth
 export interface PasswordSettings {
```
Comment (thaddmt) [inline packages/ui/src/types/authenticator/form.ts:81]: Made a similar change in my PR, what do you think about?

File: packages/ui/src/machines/authenticator/actions.ts [hunk 2]
```diff
@@ -155,18 +155,14 @@ export const setUnverifiedContactMethods = assign({
...
-    /**
-     * @migration event.data was the fallback here,
-     *  setting the entire event.data as user
-     */
-    return { ...event.data };
+    if (event.data.isSignedIn) {
+      return await Auth.getCurrentUser();
+    } else {
+      return { ...event.data };
+    }
```
Comment (thaddmt) [inline packages/ui/src/machines/authenticator/actions.ts:165]: Makes sense to me. Have we run into the case where this was called when the user was already signed in?

Comment (ioanabrooks) [reply packages/ui/src/machines/authenticator/actions.ts:165]: Yep, setUser is called quite frequently and getCurrentUser errors out if user is already signed in.

Comment (thaddmt) [reply packages/ui/src/machines/authenticator/actions.ts:165]: Discussed offline, this is not needed for now

**Review by esauerbo — APPROVED** [review]
lgtm!

**Review by thaddmt — APPROVED** [review]
LGTM!

**Pull request merged**